lib/node-loader-babel.js

```diff
@@ -43,9 +51,9 @@ export async function load(url, context, defaultLoad) {
 
     return {
       source: transformed.code,
-      // Maybe a shaky assumption
-      // TODO: look at babel config to see whether it will output ESM/CJS or other formats
-      format: "module",
+      // NOTE: transform-modules-commonjs doesn't work properly.
+      //       We put a branch here just for consistency.
+      format: transformed.sourceType === "module" ? "module" : "commonjs",
```

Comment:
Does `transformed.sourceType` really reflect the output format? Perhaps it makes more sense to assume output is `module` or fail if it is not given `source` is not applicable for `format: "commonjs"` (see https://nodejs.org/api/esm.html#loadurl-context-defaultload).

Reply:
I don't know why `transformed.sourceType` would not reflect the real output format. Are there bugs in babel or common plugins that make this wrong?

Reply:
> I don't know why `transformed.sourceType` would not reflect the real output format. Are there bugs in babel or common plugins that make this wrong?

I do not know, I am just confused as to whether `sourceType` corresponds to the input `sourceType` or the output `sourceType`, since source carries an input meaning to me. Experimental node loader uses the `source` keyword for both "input" and "output" but the babel equivalent is `transformed.code`. I could not find the Babel documentation that explains what `transformed.sourceType` means. If you can find it please share.